aten/src/ATen/functorch/BatchRulesScatterOps.cpp

```diff
@@ -1056,6 +1056,132 @@ std::tuple<Tensor,optional<int64_t>> masked_fill_scalar_batch_rule(
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> index_fill_int_scalar_batch_rule_impl(
+    Tensor & self, optional<int64_t> self_bdim,
+    int dim,
+    const Tensor & index, optional<int64_t> index_bdim,
+    const Scalar & value,
+    const bool inplace) {
+
+    const auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
+    Tensor self_ = moveBatchDimToFront(self, self_bdim);
+    Tensor index_ = moveBatchDimToFront(index, index_bdim);
+    dim = maybe_wrap_dim(dim, self_logical_rank);
+
+    if (inplace && !self_bdim.has_value()) {
+      vmapIncompatibleInplaceError("index_fill_");
+    }
+
+    if (!index_bdim) {
+      self_.unsqueeze_(-1);
+      self_.index_fill_(dim + 1, index_, value);
+      self_.squeeze_(-1);
```

Comment:
squeeze_ is generally "unsafe" because it depends on the shape of the Tensor. We should add an assertion that the last dimension of `self_` has shape 1 (otherwise, `squeeze_` will be a no-op and this can lead to bugs).

Reply:
I couldn't come up with a case that the last dimension of `self_` before `self_.squeeze_(-1)` performed has shape other than `1`. 

`self_.unsqueeze_(-1)` makes the last dimension have shape `1` even the `self_`'s shape is `[]` (which will become `tensor([], size=(0, 1))`), and `self_.index_fill_` doesn't change itself's shape. As a result, we always have the last dimension -1 to squeeze.

Maybe I'm missing something, would love to hear and add an assertion if there is a case.

Reply:
I will add a `self_logical_rank == 0` check right before unsqueeze and squeeze are performed.